### PR TITLE
tests(ticdc): fix unstable TestWriterRedoGC

### LINKED ISSUE #5736: Flaking test TestWriterRedoGC
### Which jobs are flaking?

verify

### Which test(s) are flaking?

TestWriterRedoGC

### Jenkins logs or GitHub Actions link

```log
[2022-06-03T07:49:47.594Z] === Failed

[2022-06-03T07:49:47.594Z] === FAIL: cdc/redo/writer TestWriterRedoGC (0.01s)

[2022-06-03T07:49:47.594Z]     writer_test.go:760: 

[2022-06-03T07:49:47.594Z]         	Error Trace:	writer_test.go:760

[2022-06-03T07:49:47.594Z]         	Error:      	Should have called with given arguments

[2022-06-03T07:49:47.594Z]         	Test:       	TestWriterRedoGC

[2022-06-03T07:49:47.594Z]         	Messages:   	Expected "GC" to have been called with:

[2022-06-03T07:49:47.594Z]         	            	[mock.Anything]

[2022-06-03T07:49:47.594Z]         	            	but actual calls were:

[2022-06-03T07:49:47.594Z]         	            	        []

[2022-06-03T07:49:47.594Z]         	            	[]

[2022-06-03T07:49:47.594Z]         	            	[]

[2022-06-03T07:49:47.594Z]         	            	[]

[2022-06-03T07:49:47.594Z] 

[2022-06-03T07:49:47.594Z] DONE 2495 tests, 1 failure in 527.208s

[2022-06-03T07:49:50.121Z] make: *** [unit_test_in_verify_ci] Error 1

```

### Anything else we need to know

- Does this test exist for other branches as well?

- Has there been a high frequency of failure lately?

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: close #5736 

### What is changed and how it works?
Wait for gc to be invoked before closing `Writer`

#### Tests 

 - Unit test

```release-note
`None`.
```

### CODE REVIEW

**Review by hi-rustin — APPROVED** [review]

File: cdc/redo/writer/writer_test.go [hunk 1]
```diff
@@ -751,7 +751,7 @@ func TestWriterRedoGC(t *testing.T) {
 			cfg:       cfg,
 		}
 		go writer.runGC(context.Background())
-		time.Sleep(time.Duration(defaultGCIntervalInMs+1) * time.Millisecond)
+		time.Sleep(time.Duration(defaultGCIntervalInMs+10) * time.Millisecond)
```
Comment (amyangfei) [inline cdc/redo/writer/writer_test.go:754]: Prefer to add a callback function in `mockWriter.On("GC", mock.Anything).Return(nil)`, wait for the condition to be true and check GC has ran as expected.
such as
```diff
+               gcRunCounter := atomic.NewInt32(0)
                if tt.args.isRunning {
-                       mockWriter.On("GC", mock.Anything).Return(nil)
+                       mockWriter.On("GC", mock.Anything).Return(nil).Run(func(args mock.Arguments) {
+                               gcRunCounter.Inc()
+                       })
                }
                writer := LogWriter{
                        rowWriter: mockWriter,
@@ -751,13 +755,18 @@ func TestWriterRedoGC(t *testing.T) {
                        cfg:       cfg,
                }
                go writer.runGC(context.Background())
-               time.Sleep(time.Duration(defaultGCIntervalInMs+1) * time.Millisecond)

                writer.Close()
                mockWriter.AssertNumberOfCalls(t, "Close", 2)

                if tt.args.isRunning {
-                       mockWriter.AssertCalled(t, "GC", mock.Anything)
+                       require.Eventually(t, func() bool {
+                               if gcRunCounter.Load() != int32(2) {
+                                       return false
+                               }
+                               mockWriter.AssertCalled(t, "GC", mock.Anything)
+                               return true
+                       }, time.Second, time.Millisecond*1)
```

**Review by amyangfei — APPROVED** [review]

**Pull request merged**